piksi_tools/bootload.py

```diff
@@ -77,23 +77,26 @@ def _handshake_callback(self, sbp_msg):
                         MsgBootloaderHandshakeDevice(sbp_msg).flags & 0xF)
     self.handshake_received = True
 
-  def wait_for_handshake(self, timeout=None):
+  def handshake(self, timeout=None):
     if timeout is not None:
       t0 = time.time()
     self.handshake_received = False
+    expire = time.time() + 15.0
```

Comment:
You may consider cherry picking piksi_tools.timeout.Timeout out of my bootlooader-tests PR and using that here.